### PR TITLE
Add Git pre-push hook

### LINKED ISSUE #4: Add Git hooks
Set up Git hooks to enforce code quality. Prior to pushing any commits, there should be a hook that checks for the following:
- Coding style adheres to Checkstyle
- Program is able to build and run
- Program passes all tests

In order to not have to manually set up the Git hooks locally or to add `.git/hooks` to the working tree, the [Gradle Git Hook](https://plugins.gradle.org/plugin/com.star-zero.gradle.githook) library will be used. The relevant permission request thread is [here](https://github.com/nus-cs2103-AY2021S1/forum/issues/201).

### PR DESCRIPTION
**Pull request by ianyong**
Resolves #4.

Unfortunately, the [Gradle Git Hook](https://plugins.gradle.org/plugin/com.star-zero.gradle.githook) library has compatibility issues with Windows. As such, we are currently using [git-hooks-gradle-plugin](https://plugins.gradle.org/plugin/com.github.jakemarsden.git-hooks).

Actions to be taken before this PR can be merged:
- [x] Testing by @zhaojj2209 or @siddarth2824 to verify that it works on Unix systems
- [x] Request for permission to use the new library (once the previous action is taken)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -47,6 +48,14 @@ pitest {
     threads = 4
 }
 
+gitHooks {
+    hooksDirectory = file('.git/hooks')
+    gradleCommand = './gradlew'
+    hooks = [
+        'pre-push': 'clean check test'
```
Comment (ianyong) [inline build.gradle:55]: `pitest` can be added at a later time when we get around to achieving full coverage.

Comment (wltan) [inline build.gradle]: Is it necessary to perform `test` after `check`? I believe `check` already includes `test`.

https://stackoverflow.com/questions/50104666/gradle-difference-between-test-and-check

Comment (ianyong) [reply build.gradle]: My bad, removed `test`.

**Review by zhaojj2209 — APPROVED** [review]
Tested with macOS and works!

**Comment by ianyong** [comment]
Requested for permission to use the external library at nus-cs2103-AY2021S1/forum#253.

**Review by wltan — COMMENTED** [review]
Hook does not work on my system currently.

image

I suspect that this is due to an installation of Cygwin preferring the shell script over the batch file and then getting stuck at the carriage return characters. Let me try to figure something out.

**Review by wltan — APPROVED** [review]
Managed to get it to work by converting `gradlew` from CRLF to LF locally. If there are any further problems, I'll try to settle it on my own.

LGTM.

**Pull request merged**